src/_guides/language/sound-dart.md

```diff
@@ -1,54 +1,51 @@
+title: Dart's Type System
 description: Why and how to write sound Dart code.
 ---
 <?code-excerpt replace="/([A-Z]\w*)\d\b/$1/g; /\b(main)\d\b/$1/g"?>
 
-{% comment %}
-update-for-dart-2
-{% endcomment %}
+Dart has a sound static
+type system that uses a combination of static and [runtime checks](#runtime-checks) to
+ensure that your code is type safe: runtime types always match static types.
+Because Dart has [type inference](#type-inference),
+you don't always have to specify types.
 
-This guide tells you why and how to write sound (type safe) Dart code.
-You'll learn how to use strong mode to enable soundness, as well as
-how to substitute types safely when overriding methods.
-
-**Strong mode** is a sound static
-type system that uses a combination of static and runtime checks to
-ensure your code is type safe&mdash;that you can never see a value
-whose runtime type does not match its static type.
-With strong mode enabled (in an implementation that has both the
-static and runtime checks), Dart is a sound language.
-
-<aside class="alert alert-warning" markdown="1">
-  **Warning:**
-  Currently, the Dart dev compiler
-  ([dartdevc,]({{site.webdev}}/tools/dartdevc) also known as _DDC_)
-  is the only full implementation of strong mode.
-  VM and dart2js support are on their way.
-</aside>
+This page concentrates on the type safety features added in Dart 2.
+For a full introduction to the Dart language, including types, see the
+[language tour](/guides/language/language-tour).
 
 <aside class="alert alert-info" markdown="1">
   **Terminology note:**
-  The terms **sound** Dart, **strong mode** Dart, and **type safe** Dart
-  are sometimes used interchangeably.
-  **Classic** Dart refers to Dart before soundness was added to the language.
+  The terms **sound** Dart and **type safe** Dart
+  are often used interchangeably.
+  You might also see the term **strong mode**.
+  Strong mode was an opt-in Dart 1.x feature
+  that provided partial support for type safety.
 </aside>
 
-By writing sound Dart code today, you'll reap some benefits now,
-with more in the near future. Current benefits include finding bugs
-at compile time (rather than at runtime) using Dart's static analyzer.
-And soon you'll be able to use new tools that quickly and incrementally
-compile your sound Dart code, giving you a better overall
-developer experience.
+One benefit of type safety is the ability to find bugs
+at compile time (rather than at runtime) using Dart's [static analyzer.][analyzer]
+Another benefit is being able to quickly and incrementally
+compile Dart code.
 
-Strong mode Dart adds only a few additional rules beyond that for classic
-Dart&mdash;mostly you clarify code where the types are ambiguous or
-incorrect. In fact, most strong mode errors can be fixed by adding type
-annotations to your Lists and Maps.
+<aside class="alert alert-info" markdown="1">
+  **Note:**
+  Not all tools have been updated to fully support
+  the sound static type system.
```

Comment:
> sound static type system

&arr;

> the Dart 2 type system

(because some tools are lacking the dynamic checks)

Reply:
Wondering if we could we replace "some tools" with "dart2js" or would that still be inaccurate/wrong? DDC has the checks. I'd assume by now the VM does as well?

Reply:
I was thinking of plugins and 3rd-party tools, too.

Reply:
But actually, I can just say dart2js and things that depend on it. Simpler that way.